pkg/blackduck/ctl_blackduck_test.go

```diff
@@ -0,0 +1,1151 @@
+		ExternalPostgresPostgresAdminPassword: "",
+		ExternalPostgresPostgresUserPassword:  "",
+		PvcStorageClass:                       "",
+		LivenessProbes:                        false,
+		ScanType:                              "",
+		PersistentStorage:                     false,
+		PVCJSONSlice:                          []string{},
+		CertificateName:                       "",
+		Certificate:                           "",
+		CertificateKey:                        "",
+		ProxyCertificate:                      "",
+		Type:                                  "",
+		DesiredState:                          "",
+		Environs:                              []string{},
+		ImageRegistries:                       []string{},
+		ImageUIDMapJSONSlice:                  []string{},
+		LicenseKey:                            "",
+	}, blackduckCtl)
+}
+
+func TestGetSpec(t *testing.T) {
+	assert := assert.New(t)
+	blackduckCtl := NewBlackduckCtl()
+	assert.Equal(blackduckv1.BlackduckSpec{}, blackduckCtl.GetSpec())
+}
+
+func TestSetSpec(t *testing.T) {
+	assert := assert.New(t)
+	blackduckCtl := NewBlackduckCtl()
+	specToSet := blackduckv1.BlackduckSpec{Namespace: "test"}
+	blackduckCtl.SetSpec(specToSet)
+	assert.Equal(specToSet, blackduckCtl.GetSpec())
+
+	// check for error
+	assert.EqualError(blackduckCtl.SetSpec(""), "Error setting Blackduck Spec")
+}
+
+func TestCheckSpecFlags(t *testing.T) {
+	assert := assert.New(t)
+
+	// default case
+	blackduckCtl := NewBlackduckCtl()
+	assert.Nil(blackduckCtl.CheckSpecFlags())
+
+	// TODO: add error tests
+	var tests = []struct {
+		input    *Ctl
+		expected string
+	}{
+		// case
+		{input: &Ctl{
+			Spec:                                  &blackduckv1.BlackduckSpec{},
+			Size:                                  "notValid",
+			DbPrototype:                           "",
+			ExternalPostgresPostgresHost:          "",
+			ExternalPostgresPostgresPort:          0,
+			ExternalPostgresPostgresAdmin:         "",
+			ExternalPostgresPostgresUser:          "",
+			ExternalPostgresPostgresSsl:           false,
+			ExternalPostgresPostgresAdminPassword: "",
+			ExternalPostgresPostgresUserPassword:  "",
+			PvcStorageClass:                       "",
+			LivenessProbes:                        false,
+			ScanType:                              "",
+			PersistentStorage:                     false,
+			PVCJSONSlice:                          []string{},
+			CertificateName:                       "",
+			Certificate:                           "",
+			CertificateKey:                        "",
+			ProxyCertificate:                      "",
+			Type:                                  "",
+			DesiredState:                          "",
+			Environs:                              []string{},
+			ImageRegistries:                       []string{},
+			ImageUIDMapJSONSlice:                  []string{},
+			LicenseKey:                            "",
+		}, expected: "Size must be 'small', 'medium', 'large', or 'xlarge'"},
```

Comment:
@rrati Do you think these tests should check for specific error messages or just if an error is thrown?